app/src/main/java/org/schabi/newpipe/player/helper/PlaybackParameterDialog.java

```diff
@@ -268,6 +345,42 @@ private void setupSkipSilenceControl(@NonNull final View rootView) {
         }
     }
 
+    private void setupAdjustBySemitonesControl(@NonNull final View rootView) {
+        adjustBySemitonesCheckbox = rootView.findViewById(R.id.adjustBySemitonesCheckbox);
+        if (adjustBySemitonesCheckbox != null) {
+            // restores whether semitone adjustment is used or not
+            adjustBySemitonesCheckbox.setChecked(PreferenceManager
+                .getDefaultSharedPreferences(requireContext())
+                .getBoolean(getString(R.string.playback_adjust_by_semitones_key), true));
+
+            // stores whether semitone adjustment is used or not
+            adjustBySemitonesCheckbox.setOnCheckedChangeListener((compoundButton, isChecked) -> {
+                PreferenceManager.getDefaultSharedPreferences(requireContext())
+                    .edit()
+                    .putBoolean(getString(R.string.playback_adjust_by_semitones_key), isChecked)
+                    .apply();
+                togglePitchSliderType(rootView);
+                if (isChecked) {
+                    setPlaybackParameters(
+                            getCurrentTempo(),
+                            getCurrentPitch(),
+                            percentToSemitones(getCurrentPitch()),
+                            getCurrentSkipSilence()
+                    );
+                    setSemitoneSlider(percentToSemitones(getCurrentPitch()));
```

Comment:
I found a small issue: when you set the pitch to 300% and then turn on the "Adjust by semitones" checkbox, the semitones are set to `19`, but if you press the checkbox again the pitch is set to 200% and the video pitch actually changes. What I expected to happen when pressing the checkbox the first time was for the semitones to be trimmed down to `12`.
```suggestion
                    setSemitoneSlider(max(-12, min(12, percentToSemitones(getCurrentPitch()))));
```

Reply:
I though about both options, but I guess it really makes sense to trim it down to 12.